wandb/cli.py

```diff
@@ -445,6 +447,106 @@ def sync(ctx, path, id, project, entity, ignore):
+    elif "://" in path:
+        wandb.termlog("Logging reference artifact from {path} to: \"{artifact_path}\" ({type})".format(
+            path=path, type=type, artifact_path=artifact_path))
+        artifact.add_reference(path)
+    else:
+        raise ClickException("Path argument must be a file or directory")
+
+    run = wandb.init(entity=entity, project=project, config={"path": path}, job_type="cli_put")
+    # We create the artifact manually to get the current version
+    res = api.create_artifact(type, artifact_name, artifact.digest,
+        entity_name=entity, project_name=project, run_name=run.id, description=description,
+        aliases=[{"artifactCollectionName": artifact_name, "alias": a} for a in alias])
+    artifact_path = artifact_path.split(":")[0] + ":" + res.get("version", "latest")
+    # Re-create the artifact and actually upload any files needed
+    run.log_artifact(artifact, aliases=alias)
+    wandb.termlog("Artifact uploaded, use this artifact in a run by adding:\n", prefix=False)
+
+    wandb.termlog("    artifact = run.use_artifact(\"{path}\")\n".format(
+        path=artifact_path,
+        type=type
+    ), prefix=False)
+
+
+@artifact.command(context_settings=CONTEXT, help="Download an artifact from wandb")
+@click.argument("path")
+@click.option("--root", help="The directory you want to download the artifact to")
+@click.option("--type", help="The type of artifact you are downloading")
+@display_error
+def get(path, root, type):
+    entity, project, artifact_name = public_api._parse_artifact_path(path)
+    if project is None:
+        project = click.prompt("Enter the name of the project you want to use")
+
+    try:
+        artifact_parts = artifact_name.split(":")
+        if len(artifact_parts) > 1:
+            version = artifact_parts[1]
+            artifact_name = artifact_parts[0]
+        else:
+            version = "latest"
+        full_path = "{entity}/{project}/{artifact}:{version}".format(
+            entity=entity, project=project,
+            artifact=artifact_name, version=version)
+        wandb.termlog("Downloading {type} artifact {full_path}".format(
+            type=type or "dataset", full_path=full_path))
+        artifact = public_api.artifact(full_path, type=type)
+        path = artifact.download(root=root)
+        wandb.termlog("Artifact downloaded to %s" % path)
+    except ValueError:
+        raise ClickException("Unable to download artifact")
+
+@artifact.command(context_settings=CONTEXT, help="List all artifacts in a wandb project")
+@click.argument("path")
+@click.option("--type", "-t", help="The type of artifacts to list")
+@display_error
+def ls(path, type):
+    if type is not None:
+        types = [public_api.artifact_type(type, path)]
+    else:
+        types = public_api.artifact_types(path)
+
+    def human_size(bytes, units=['','KB','MB','GB','TB', 'PB', 'EB']):
+        return str(bytes) + units[0] if bytes < 1024 else human_size(bytes>>10, units[1:])
```

Comment:
I think thought we had a util function for this. Maybe not.

Reply:
Currently we always show Megabytes.